File: test/gen_js_client/hello-message.js
var baseAPI = require('rapier')

var exports = function() {
            
    function API() {}
    
    API.prototype = Object.create(baseAPI.BaseAPI.prototype);
    API.prototype.well_known_URLs = function() {
        return ['/message']
    }
    API.prototype.resourceClass = function(type_name) {
        return  type_name in classToKindMap ? classToKindMap[type_name] : baseAPI.BaseResource      
    }
    
    var api = new API();

    var api_function = function() {
        return api
    }

    function HelloMessage(jso, url, etag) {
        baseAPI.BaseEntity.call(this, jso, url, etag)
    }
    HelloMessage.prototype = Object.create(baseAPI.BaseEntity.prototype);
    HelloMessage.prototype.constructor = HelloMessage;
    HelloMessage.prototype._className = 'HelloMessage';
    HelloMessage.prototype.api = api_function;

    function Collection(jso, url, etag) {
        baseAPI.BaseEntity.call(this, jso, url, etag)
    }
    Collection.prototype = Object.create(baseAPI.BaseCollection.prototype);
    Collection.prototype.constructor = Collection;
    Collection.prototype._className = 'Collection';
    Collection.prototype.api = api_function;

    var classToKindMap = {
        HelloMessage: HelloMessage,
        Collection: Collection
        }

    return {
        api: api,
        HelloMessage: HelloMessage,
        Collection: Collection
        }
        
}
    
module.exports = exports()

File: util/test/gen_py_sdk/hello-message.py

from rapier.py.base_api import BaseAPI, BaseResource, BaseEntity, BaseCollection

class API(BaseAPI):
    def well_known_URLs(self):
        return ['/message']
    def resource_class(self, type_name):
        return classToKindMap.get(type_name, BaseResource)

api = API()

class APIClass(object):
    def api(self):
        return api

class HelloMessage(BaseEntity, APIClass):            
    pass

class Collection(BaseCollection, APIClass):            
    pass

classToKindMap = {
    'HelloMessage': HelloMessage,
    'Collection': Collection
    }
